packages/custom-functions-metadata-plugin/src/customfunctionsplugin.ts

```diff
@@ -23,24 +27,29 @@ class CustomFunctionsMetadataPlugin {
         const inputFilePath = path.resolve(this.options.input);
         let generateResult: IGenerateResult;
 
-        compiler.hooks.compile.tap(pluginName, async () => {
+        compiler.hooks.beforeCompile.tapPromise(pluginName, async () => {
             generateResult = await generateCustomFunctionsMetadata(inputFilePath, true);
         });
 
-        compiler.hooks.emit.tap(pluginName, (compilation) => {
+        compiler.hooks.compilation.tap(pluginName, (compilation) => {
             if (generateResult.errors.length > 0) {
                 generateResult.errors.forEach((err: string) => compilation.errors.push(new WebpackError(inputFilePath + " " + err)));
             } else {
                 compilation.assets[this.options.output] = new sources.RawSource(generateResult.metadataJson);
+                if (!global.generateResults) global.generateResults = {};
```

Comment:
Let's indent this to be like:
```
if (!global.generateResults) { 
    global.generateResults = {};
}
```
To be more consistent with the rest of the codebase.